### ISSUE TITLE
Thoughts about timers and countdowns

### ISSUE THREAD

**Issue by interrogator** [uzh/marugoto#64]
Please note that this is a content only issue, not code. Mostly for @martindusinberre @Helena-LiT . And it's just food for thought.

I am just thinking about the genre of Plantation Lives. It occurs to me that timed elements of the game, where the user has to rush to do something, might actually be a violation of a principle of text adventure games. This game is obviously more than simple text adventure, but the point is still, the game is strategy, not action, and exploration rather than racing. Having timers, countdowns etc. can work against this spirit. This game is a world-exploring one, and I think it's pretty rare to find countdowns in these kinds of games.

That said, I think such things can be quite fun, and definitely add something cool to the game. So what I'd like to see, is a PL where timed parts of the game are somehow flagged pretty clearly, and perhaps even on the preceding page (`Careful ... once you hit Continue, you'll have to work fast!`). And also, just double checking that any timer actions (phone call occurring in the middle of a page) could not be just as nicely placed in places where the user triggers them.

Personally, I don't like the chapter 1 part where you need to take notes on the painting as fast as you can, because soon time will be up. The user doesn't know here if they have 5 seconds or 5 hours, and nothing on the page is moving (like a progress bar or something, which helps users know that something is happening in the background). I would tell the user how much time they actually have, either by displaying the clock or at least through something clever like `Quick! The session chair has just raised the One Minute Left sign!`. Without knowing approximately how much time there is to complete the note-taking, users really have no idea if they are expected to write a bullet-point or a whole paper.

This is the kind of game people might like to play slowly, at their own pace, and therefore it'd be nice if users could kind of control when the enter time-critical spaces in game.

By the way, how have our test users responded to timers? Anybody said they particularly liked/didn't like a timed part?

**Comment by l0rb** [comment]
I do agree that time based events feel out of place for the genre. They also do inhibit self-pacing of the player/learner. A possible compromise would remove all effects of timers on the game-world and progress in the game. There could still be timers but if a player runs out of time they can still finish the task with the consequences only affecting things that are irrelevant to the game per se. If there is a final report or score in the game that would be a good place to show what timers have been missed.

**Comment by interrogator** [comment]
For context, @l0rb is a friend and former colleague of mine. I reached out to him to test the game because he is a tremendously bright young man, a gamer, and a big fan of open source. Welcome @l0rb.

**Comment by martindusinberre** [comment]
Great, many thanks. 
The rationale for the timing function is two-fold:
i) it actually replicates a big reality of being an academic (not just a historian). 
ii) in this particular module (Plantation Lives), it's a way of the player understanding viscerally one of the biggest elements of being a plantation worker, namely the unbending new time regimes at work on the plantation. In this way, it's a small nod towards "experiencing" plantation work without us going into the genre of "pretend you're a plantation worker" (this second point came out of the proof of concept test back in September 2017)
As for feedback, it's overwhelmingly positive on the timing functionality--people love that it puts them under pressure and it's often the emotional response they have--"OMG, I only have five minutes"--that they recall from the game, rather than the content. That's one of the ways we want it to be, that they *feel* what it is to be a historian, rather than just "role-play" it.
The only feedback is about *where* the timed exercises occur. I agree that it doesn't work ideally in chapter 1, and am planning to change that. Note also the "timerVisible" functionality (in each page.json) which allows the player to see the countdown to how long they have left on an exercise--this we can use more, I think.
In any case, chapter 1 needs some tweaking, that's completely clear. Any other feedback from @l0rb greatly appreciated. Thanks, @interrogator

**Comment by interrogator** [comment]
Great discussion, team. It's good to hear that players enjoy the time-critical components, evene if that's a bit contra to my intutions! But yes, having these time constraints is definitely something that differentiates PL from classic text-adventure games (i.e. [Colossal Cave](https://en.wikipedia.org/wiki/Colossal_Cave_Adventure) or other world-exploration games).

The compromise here is reasonably clear then ... just to make sure time pressure is applied purposefully, and with `timerVisible`, or any other clues about approximate amounts of time, whenever it makes sense. Would be good to keep monitoring players' thoughts about this issue, given that @l0rb and I had very different feelings to other testers.

**Comment by martindusinberre** [comment]
Good, do let's keep monitoring this. Thanks so much again for your inputs.
@l0rb please feel free to join us on other stuff, and follow us also on Twitter :) @TransitLives

**Comment by Helena-LiT** [comment]
This is not how I remember the players's reactions to the timinig functionality. I have checked our feedback data for the Basel workshop and the first student usability test - the experience was mostly negative. In Basel (ca. 10 testers) there were 3 users who marked it as a negative and only one who liked it; in Zurich (10 testers) there were also 4 users who complained about how the game handles time.
![TimePressure](https://user-images.githubusercontent.com/53467834/65488067-89d7bd00-dea8-11e9-9f0f-1eb034a98ed5.JPG)
I do agree that for this storyline the time pressure feature makes sense, but I would avoid it in my own storyline. Personally, I agree with @l0rb and @interrogator that the game feels like it should be more about strategy and exloration rather than emotional stress.

**Comment by interrogator** [comment]
Wow, nice to have some concrete user data, that definitely helps! I guess we can ultimately do something like summarise some tips for game creation for people wishing to make games. There we could make a note about the pros and cons of time pressure...

**Comment by martindusinberre** [comment]
I was going by the great chart @Helena-LiT  developed which responded to how people felt about playing in November 2017, in which issues 03, 04, 05, 06 and 07 all referenced the reality of playing the game--and I take time (perhaps mistakenly) to be a key aspect of this. 
We also made the decision to make time a bigger part of Plantation Lives than other modules exactly in order to deal with (ii) above. The Zurich feedback I completely agree with but think it refers to the timing of the timing, as it were--that is, where we have timed exercises, and how.
Am very happy to revisit this in future modules, but would caution against cutting time pressure completely--it's one of the key realities of academic life. And evidenced by my trying to write this in a tremendous rush before going off to class :)
Thanks all
